File: backend/src/main/java/com/musicmatch/backend/dto/UserProfileResponse.java
package com.musicmatch.backend.dto;

import com.musicmatch.backend.model.ExperienceLevel;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
public class UserProfileResponse {

    private Long id;
    private String username;

    private String biography;

    private Long cityId;
    private String cityName;

    private ExperienceLevel experienceLevel;

    private List<Long> styleIds;
    private List<InstrumentLevelResponse> instruments;

    private String profilePicture;
}

File: mobile/app/src/main/java/com/musicmatch/mobile/model/dto/UserProfileResponse.kt
package com.musicmatch.mobile.model.dto

import com.musicmatch.mobile.model.ExperienceLevel

data class UserProfileResponse(
    val id: Long,
    val username: String,
    val biography: String?,
    val cityId: Long?,
    val cityName: String?,
    val profilePicture: String?,
    val experienceLevel: ExperienceLevel?,
    val styleIds: List<Long>?,
    val instruments: List<InstrumentLevelResponse>?
)